module-3/lab-intro-to-ml/your-code/main.ipynb

```diff
@@ -12,7 +12,7 @@
   },
```

Comment:
- It's better to not include any hard-code data in your code:
```Python
124 / len(austin["PrecipitationSumInches"])
#better if you do something similar to the following.
missing_values['PrecipitationSumInches'].size / austin['PrecipitationSumInches'].size
```
- When possible as less lines of code as possible:
```Python
# Your code here
austin_fixed["Rain"] = austin_fixed["Events"].str.contains("Rain").astype(int)
austin_fixed["Snow"] = austin_fixed["Events"].str.contains("Snow").astype(int)
austin_fixed["Fog"] = austin_fixed["Events"].str.contains("Fog").astype(int)
austin_fixed["Thunderstorm"] = austin_fixed["Events"].str.contains("Thunderstorm").astype(int)
#you can use a simple for loop
for event in event_list:
    austin_fixed[event] = austin_fixed['Events'].str.contains(event).astype(int)
```
- When having quite simple functions, you can try to use lambda functions
``Python
def to_date(x):
    d = dt.strptime(x,'%Y-%m-%d')
    return d.toordinal()
```
- Why have you skipped the sampling step in challenge 5?
- Bonus Challenge 2: Here try to train a linear SVM instead of the rbf one you've used. Previous to training a classifier it's a good idea to plot a scatter plot of your data for each label/class. This can give you a visual idea of how well a linear or not linear classifier will perform in the training step. 
```Python
# Your code here
clf = svm.SVC(gamma="auto")
clf.fit(X_train,y_train)
clf.score(X_test,y_test)
```
- If you want to play with synthetic data for classification problems, scikit-learn has a function to generate synthetic datasets. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html.
It's very useful to play with different type of classifiers (two label)
```Python
from sklearn.datasets import make_classification
```